client/order/bookside.go

```diff
@@ -106,6 +108,23 @@ func (d *bookSide) Remove(order *Order) error {
 	return fmt.Errorf("order %s not found", order.OrderID)
 }
 
+// UpdateRemaining updates the remaining quantity for an order. If the order is
+// found it will be returned, else nil.
+func (d *bookSide) UpdateRemaining(oid order.OrderID, remaining uint64) *Order {
+	d.mtx.Lock()
+	defer d.mtx.Unlock()
+
+	for _, bin := range d.bins {
+		for _, ord := range bin {
+			if ord.OrderID == oid {
+				ord.Quantity = remaining
```

Comment:
Might also consider keeping separate fields for originnal quantity and remaining quantity, but I haven't yet seen a need.